src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,116 +1,105 @@
-import React, { useState } from 'react';
-import { postComment } from '../../api/comments';
-
 import './NewCommentForm.scss';
-
-import { Loader } from '../Loader';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../react-app-env';
 
 type Props = {
-  postId: number,
-  fetchComments: () => void,
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
 };
 
-export const NewCommentForm: React.FC<Props> = React.memo(({
-  postId,
-  fetchComments,
+export const NewCommentForm: React.FC<Props> = ({
+  commentsList, setCommentsList,
 }) => {
-  const [newCommentName, setNewCommentName] = useState('');
-  const [newCommentEmail, setNewCommentEmail] = useState('');
-  const [newCommentBody, setNewCommentBody] = useState('');
-  const [isInputFill, setIsInputFill] = useState(false);
-  const [isLoading, setLoading] = useState(false);
-
-  const removeFillError = () => {
-    if (isInputFill) {
-      setIsInputFill(false);
-    }
-  };
-
-  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setNewCommentName(event.target.value);
-    removeFillError();
-  };
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
 
-  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setNewCommentEmail(event.target.value);
-    removeFillError();
-  };
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
```

Comment:
No need in this variable, let's use it right away
```suggestion
      setCommentsList([ ...commentsList ]);
```